src/nodes.coffee

```diff
@@ -949,6 +949,28 @@ exports.StringLiteral = class StringLiteral extends Literal
     unquoted = unquoted.replace /\\n/g, '\n' if csx
     unquoted
 
+  withoutQuotesInLocationData: ->
```

Comment:
This is more of the "actual string literal" vs "text chunk inside interpolation" business

Basically, a `TaggedTemplateLiteral` always wants its "string" to be a `StringWithInterpolations`, but if it's a simple non-interpolated string eg `a"b"`, it'll have been parsed as a `StringLiteral` so a conversion from `StringLiteral` -> `StringWithInterpolations` is required

And the existing version of that conversion didn't assign accurate location data

Reply:
I feel like some of this explanation should probably go in comments in the code? Like a reminder of why this method exists and when one should use it?

Reply:
Sure, added an explanatory comment